src/app/_validation/PersonValidation.ts

```diff
@@ -1,12 +1,16 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
 import { PersonInput } from '../../interfaces';
-export const isValidFirstName = (firstName: string): boolean => firstName?.length > 0;
+export const isValidFirstName = (): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null =>
+        control.value?.length > 0
```

Comment:
interesting place, 
let's think for a second, we're taking input from user and checking using `control.value?.length`

may I ask you what are the downsides of this approach what value could later break saving process to DB, for example